lib/ClangImporter/ImportType.cpp

```diff
@@ -1049,7 +1045,7 @@ namespace {
 
         importedType = ProtocolCompositionType::get(Impl.SwiftContext,
                                                     members,
-                                                    /*HasExplicitAnyObject=*/false);
+                                                    /*HasExplicitAnyObject=*/hasNSObjectProtocol);
```

Comment:
This doesn't seem so important when `members` is known to contain at least one `@objc` protocol. Or does this handle the 0 case?

Reply:
It's only really relevant for the 0 case; I can gate it on that.

Reply:
FWIW, this is the part that forces us to import such things as `AnyObject` rather than `Any`; @jckarter was arguing that we should import as `Any`.